src/plugins/cloudMetadata.nim

```diff
@@ -197,23 +206,33 @@ proc getAwsToken(): Option[string] =
   trace("Retrieved AWS metadata token")
   return some(body)
 
-template oneItem(keyname: string, url: string) =
+proc oneItem(chalkDict: ChalkDict, token: string, keyname: string, url: string) =
+  ## If `keyname` is subscribed, hits the given `url` and sets the `keyname` key
+  ## in `chalkDict` to the value of the response (if non-empty).
   if isSubscribedKey(keyname):
     let
       hdrs      = newHttpHeaders([("X-aws-ec2-metadata-token", token)])
       resultOpt = hitProviderEndpoint(url, hdrs)
     if resultOpt.isSome():
-      setIfNotEmpty(result, keyname, resultOpt.get())
+      setIfNotEmpty(chalkDict, keyname, resultOpt.get())
 
-template listKey(keyname: string, url: string) =
+proc listKey(chalkDict: ChalkDict, token: string, keyname: string, url: string) =
+  ## If `keyname` is subscribed, hits the given `url` and sets the `keyname` key
+  ## in `chalkDict` to the value of the response (as a `seq` of lines).
   if isSubscribedKey(keyname):
     let
       hdrs      = newHttpHeaders([("X-aws-ec2-metadata-token", token)])
       resultOpt = hitProviderEndpoint(url, hdrs)
     if resultOpt.isSome():
-      setIfNeeded(result, keyname, resultOpt.get().splitLines())
-
-template jsonKey(keyname: string, url: string) =
+      setIfNeeded(chalkDict, keyname, resultOpt.get().splitLines())
```

Comment:
nit

```suggestion
      chalkDict.setIfNeeded(keyname, resultOpt.get().splitLines())
```

since you refactored other places like that in the same module. I think there are a few left as well